search-task/src/pages/homepage/index.js

```diff
@@ -105,7 +110,7 @@ const Homepage = () => {
             {Array.isArray(data) &&
               data.length > 0 &&
               data?.map((data) => (
-                <Card key={data.char_id} name={data.name} image={data.img} />
+                <Card key={data.char_id} name={data.name} image={data.img} rdata={data}/>
```

Comment:
Just pass the data prop in component, remove other fields like name, image, rdata and stuffs

Reply:
try to make variable names specific to their action